doc/use_cases.rst

```diff
@@ -0,0 +1,113 @@
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+###############
+ DNF Use Cases
+###############
+
+.. contents::
+
+==============
+ Introduction
+==============
+
+Every feature present in DNF should be based on a reasonable use case. All the
+supported use cases are supposed to be enumerated in this document.
+
+In case you use DNF to achieve a goal which is not documented here, either you
+found an error in the documentation or you misuse DNF. In either case we would
+appreciate if you share the case with us so we can help you to use DNF in the
+correct way or add the case to the list. You can only benefit from such a
+report because then you can be sure that the behavior that you expect will not
+change without prior notice in the :doc:`release_notes` and that the behavior
+will be covered by our test suite.
+
+.. IMPORTANT::
+
+  Please consult every usage of DNF with our reference documentation to be sure
+  what are you doing. The examples mentioned here are supposed to be as simple
+  as possible and may ignore some minor corner cases.
+
+.. WARNING::
+
+  The list is not complete yet - the use cases are being added incrementally
+  these days.
+
+=====================
+ General assumptions
+=====================
+
+DNF must always preserve the integrity of the related RPMDB.
+
+The user in question must have the appropriate permissions.
+
+.. _install_use_case-label:
+
+========================================================================================
+ Ensure that my system contains given mix of features (packages/files/providers/groups)
+========================================================================================
+
+A system administrator has a list of features that has to be present in an
+operating system. The features must be provided by RPM packages in system
+repositories that must be accessible.
+
+A feature may be for example a concrete version of a package
+(``hawkey-0.5.3-1.fc21.i686``), a pathname of a binary RPM file
+(``/var/lib/mock/fedora-21-i386/result/hawkey-0.5.3-2.20150116gitd002c90.fc21.i686.rpm``),
+an URL of a binary RPM file
+(``http://jenkins.cloud.fedoraproject.org/job/DNF/lastSuccessfulBuild/artifact/fedora-21-i386-build/hawkey-0.5.3-99.649.20150116gitd002c90233fc96893806836a258f14a50ee0cf47.fc21.i686.rpm``),
+a configuration file (``/etc/yum.repos.d/fedora-rawhide.repo``), a language
+interpreter (``ruby(runtime_executable)``), an extension (``python3-dnf``), a
+support for building modules for the current running kernel
+(``kernel-devel-uname-r = $(uname -r)``), an executable (``*/binaryname``) or a
+collection of packages specified by any available identifier (``kde-desktop``).
```

Comment:
No, the symbol is intentionally missing because the name does not contain the symbol and ```group_by_pattern``` wouldn't understand it. The character is understood only on the CLI level and it's included in the example already.